wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1188,6 +1382,59 @@ instance BuildableSafeGen Transaction where
 instance Buildable [Transaction] where
     build = bprint listJson
 
+-- | Signature of transaction. We are use it in case of
+-- externally signed transaction (mobile client or hardware wallet).
+data TransactionSignature = TransactionSignature
+  { etxSignature :: !Text -- Should it be 'Text'?
+  } deriving (Show, Ord, Eq, Generic)
+
+deriveJSON Serokell.defaultOptions ''TransactionSignature
+
+instance ToSchema TransactionSignature where
+  declareNamedSchema =
+    genericSchemaDroppingPrefix "etx" (\(--^) props -> props
+      & ("signature" --^ "Raw signature of new transaction.")
+    )
+
+instance Arbitrary TransactionSignature where
+  arbitrary = TransactionSignature <$> arbitrary
+
+deriveSafeBuildable ''TransactionSignature
+instance BuildableSafeGen TransactionSignature where
+    buildSafeGen sl TransactionSignature{..} = bprint ("{"
+        %" signature="%buildSafe sl
+        %" }")
+        etxSignature
+
+-- | A 'Wallet''s 'SignedTransaction'. It is assumed
+-- that this transaction was signed on the client-side
+-- (mobile client or hardware wallet).
+data SignedTransaction = SignedTransaction
+  { stxTransaction :: !Transaction
+  , stxSignature   :: !TransactionSignature
```

Comment:
I think we might need to do something a bit more sophisticated here, as (probably) what we will be passed externally is just a CBOR (signed) blob plus some witness indeed -- I think the best course of action is to take a look at what we thought originally with Duncan & Dmitry back in November (see [here](https://rebilly.github.io/ReDoc/?url=https://gist.githubusercontent.com/adinapoli-iohk/93791a963a9eb545315840c784fab3c2/raw/bbf320febe511839ce0545d669463fc13aee1742/stub.yaml#/paths/~1api~1txs~1payments~1external/post)) and cross-reference the tech spec from Stace.

As you can see in that stub API we get passed a `txPayload`, which is just a `ByteString`, essentially.

Reply:
In this case it should be just this:

```
data SignedTransaction = SignedTransaction
  { stxTransaction :: !ByteString
  , stxSignature   :: !ByteString
```

Because `txPayload` is a binary blob, and `XSignature` is a binary blob as well. :-)